lib/charms/mongodb/v0/config_server_interface.py

```diff
@@ -106,16 +108,22 @@ def _on_relation_changed(self, event) -> None:
         # create user and set secrets for mongos relation
         self.charm.client_relations.oversee_users(None, None)
 
+        relation_data = {
+            KEYFILE_KEY: self.charm.get_secret(
+                Config.Relations.APP_SCOPE, Config.Secrets.SECRET_KEYFILE_NAME
+            ),
+            CONFIG_SERVER_DB_KEY: config_server_db,
+        }
+
+        # if tls enabled
+        int_tls_ca = self.charm.tls.get_tls_secret(
+            internal=True, label_name=Config.TLS.SECRET_CA_LABEL
+        )
+        if int_tls_ca:
+            relation_data[INT_TLS_CA_KEY] = int_tls_ca
+
         if self.charm.unit.is_leader():
```

Comment:
should we perhaps check if not-leader and `return` before line 111 (where you create relation_data)

Reply:
ah actually it isn't needed to check since its done in `pass_hooks_checks` - I will remove this